Feature: add ```--no-reload``` flag

rainDelay9 opened the pull request:
This pull request includes several suggestions to support #54:

- Change 'watch' subcommand to 'runtime' to better represent what it does. (emulates a runtime)
- Add 'watch' as a flag for runtime. (default is false)
    - Since this is a breaking change the version was bumped to 0.6.0.
    - Because this is not backwards-compatible aliasing to "start" and "watch" have been removed.

The logic behind making watch default to false, instead of a ''--no-watch" flag is this seems to align with other CLIs.
If instead we support a "--no-watch" flag then aliasing can still remain, but obviously command name needs to change, and watch as a command would be pretty silly to support as you can run

```bash
cargo lambda watch --no-watch
```

and get something valid. Start would still be ok but wouldn't behave as before.

Another option is to keep watch, and defer it to runtime with the "--watch" flag. This seems messy as now you have two commands doing virtually the same thing.

Another thing is passing manifest path and watch down the server rabbit hole. Maybe a context struct is in order.

calavera commented:
I personally don't like the name `runtime`. `Watch` aligns much better to catch-watch, which is a tool well know in the Rust ecosystem.

> The logic behind making watch default to false, instead of a ''--no-watch" flag is this seems to align with other CLIs.

Which other CLIs?

I rather add a flag for `--no-reload`, and keep the `watch` name for this command. This aligns with the behavior described in #54, and requires less code changes.

Don't change the package version, please. I don't know yet when I'll release a new minor version.

rainDelay9 commented:
Made the following changes:
- reverted command to watch
- reverted version back to 0.5.3
- added ```--no-replace``` flag as an option for watch command

calavera reviewed (approved):
Perfect, this is exactly what I was thinking, thanks a lot! 🎊 

I'm going to run a couple of local tests and will merge it if they go well.

indiv0 commented:
Thank you to both of you, this is perfect!
